src/http/curl.h

```diff
@@ -736,7 +730,41 @@ namespace ccf::curl
 
       if (self->is_stopping)
       {
-        LOG_FAIL_FMT("libuv_socket_poll_callback called while stopping");
+        LOG_FAIL_FMT(
+          "libuv_socket_poll_callback called on {} while stopped",
+          socket_context->socket);
+        return;
+      }
+
+      if (status < 0)
+      {
+        if (status == UV_EBADF)
+        {
+          // Thrown when POLLERR is thrown by the epoll socket, such as when a
+          // TCP socket received a reset at a bad time
+          // https://docs.libuv.org/en/v1.x/poll.html#c.uv_poll_start
+          // https://github.com/libuv/libuv/issues/3796
+          LOG_INFO_FMT(
+            "Socket poll error on {}: {}",
+            socket_context->socket,
+            uv_strerror(status));
+        }
+        else
+        {
+          LOG_FAIL_FMT(
+            "Socket poll error on {}: {}",
+            socket_context->socket,
+            uv_strerror(status));
+        }
+
+        // Notify curl of the error
+        CHECK_CURL_MULTI(
+          curl_multi_socket_action,
+          self->curl_request_curlm,
+          socket_context->socket,
+          CURL_CSELECT_ERR,
+          nullptr);
+        self->curl_request_curlm.perform();
         return;
```

Comment:
This seems to change the behaviour of every _other_ negative `status` as well, from log-fail-and-early-return, to notify-curl-and-perform, based on the claim that this is the right thing to do for `UV_EBADF`. Is that right? Which is the better behaviour if `status` is suddenly `-42`?